milabench/metrics/report.py

```diff
@@ -0,0 +1,159 @@
+from milabench.metrics.sqlalchemy import Pack, Exec, Metric
+
+import numpy as np
+import pandas as pd
+import sqlalchemy
+from sqlalchemy.orm import Session
+
+
+def fetch_data(client, runame):
+    stmt = (
+        sqlalchemy.select(
+            Exec.name.label("run"),
+            Pack.name.label("bench"),
+            Metric.name.label("metric"),
+            Metric.value,
+            Metric.gpu_id,
+        )
+        .join(Exec, Metric.exec_id == Exec._id)
+        .join(Pack, Metric.pack_id == Pack._id)
+        .where(Exec.name.startswith(runame))
+    )
+
+    results = []
+    with Session(client) as sess:
+        cursor = sess.execute(stmt)
+        columns = list(cursor.keys())
+
+        for row in cursor:
+            row = list(row)
+            row[0] = row[0].split(".")[0]
+            results.append(row)
+
+    df_post = pd.DataFrame(results, columns=columns)
+    return df_post
+
+
+def dropminmax(xs):
+    xs = sorted(x for x in xs if x is not None)
+    if len(xs) >= 5:
+        xs = xs[1:-1]
+    return xs
+
+
+def sem(xs):
+    xs = dropminmax(xs)
```

Comment:
Because we are making a pivot table we cannot really do the filtering on the dataframe as a whole.
all the values with different units are in the same "metric" column. the right xs vector is generated by the pivot_table and it is function of how the pivot will aggregate the series.


The dataframe looks like

```
          run    bench       metric        value gpu_id
0    sedumoje  benchio         loss     1.000000    ALL
1    sedumoje  benchio         loss     1.000000    ALL
2    sedumoje  benchio         rate  8036.001286    ALL
3    sedumoje  benchio         rate  8536.064874    ALL
4    sedumoje  benchio         loss     0.500000    ALL

133  sedumoje  benchio         rate  6351.626016    ALL
134  sedumoje  benchio     walltime     0.901159    ALL
135  sedumoje  benchio  return_code     0.000000    ALL
136  sedumoje  benchio     walltime     0.873208    ALL
137  sedumoje  benchio  return_code     0.000000    ALL
```